### PR TITLE
[AppInsights Bot] Add DialogBot and DialogAndWelcomeBot

### PR DESCRIPTION
**Pull request by gasper-az**
## Description
- Create `bots` folder
- Add `DialogBot`
- Add `DialogAndWelcomeBot`
- Add `resource` file for `DialogAndWelcomeBot`

### CODE REVIEW

File: samples/javascript_nodejs/21.corebot-app-insights/bots/dialogAndWelcomeBot.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+const { CardFactory } = require('botbuilder');
+const { DialogBot } = require('./dialogBot');
+const WelcomeCard = require('./resources/welcomeCard.json');
+
+class DialogAndWelcomeBot extends DialogBot {
+    /**
+     *
```
Comment (denscollo) [inline samples/javascript_nodejs/21.corebot-app-insights/bots/dialogAndWelcomeBot.js]: Can we remove this space? Same on `dialogbot.js` file

File: samples/javascript_nodejs/21.corebot-app-insights/bots/dialogAndWelcomeBot.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+     * @param {Dialog} dialog 
+     */
+    constructor(conversationState, userState, dialog) {
+        super(conversationState, userState, dialog);
+
+        this.onMembersAdded(async (turnContext, next) => {
+            const membersAdded = turnContext.activity.membersAdded;
+
+            for (let pos = 0; pos < membersAdded.length; pos++) {                
+
```
Comment (denscollo) [inline samples/javascript_nodejs/21.corebot-app-insights/bots/dialogAndWelcomeBot.js]: I'd remove this space

File: samples/javascript_nodejs/21.corebot-app-insights/bots/dialogAndWelcomeBot.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+                if (membersAdded[pos].id != turnContext.activity.recipient.id) {
+                    const welcomeCard = CardFactory.adaptiveCard(WelcomeCard);
+                    await turnContext.sendActivity({ attachments: [welcomeCard] });
```
Comment (denscollo) [inline samples/javascript_nodejs/21.corebot-app-insights/bots/dialogAndWelcomeBot.js:24]: And add a space between the const declaration and the awaits

File: samples/javascript_nodejs/21.corebot-app-insights/bots/dialogAndWelcomeBot.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+                    await dialog.run(turnContext, conversationState.createProperty('DialogState'));
+                }
+            }
+
```
Comment (denscollo) [inline samples/javascript_nodejs/21.corebot-app-insights/bots/dialogAndWelcomeBot.js]: Also, I think this space can be removed

File: samples/javascript_nodejs/21.corebot-app-insights/bots/dialogAndWelcomeBot.js (hunk 1 above)
Comment (gasper-az) [reply samples/javascript_nodejs/21.corebot-app-insights/bots/dialogAndWelcomeBot.js]: Feedback Applied.

File: samples/javascript_nodejs/21.corebot-app-insights/bots/dialogAndWelcomeBot.js (hunk 4 above)
Comment (gasper-az) [reply samples/javascript_nodejs/21.corebot-app-insights/bots/dialogAndWelcomeBot.js]: Feedback Applied.

**Review by denscollo — APPROVED** [review]
LGTM

**Review by Aliandi — APPROVED** [review]

**Pull request merged**